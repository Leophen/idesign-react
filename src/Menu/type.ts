export interface MenuProps {
  /**
   * 类名
   */
  className?: string;
  /**
   * 内容
   */
  children?: React.ReactNode;
  /**
   * 自定义样式
   */
  style?: React.CSSProperties;
  /**
   * 导航宽度
   */
  width?: React.CSSProperties["width"];
  /**
   * 固定选中值（受控）
   */
  active?: string | number;
  /**
   * 默认选中值（非受控）
   * @default 0
   */
  defaultActive?: string | number;
  /**
   * 前置内容
   */
  prefixContent?: React.ReactNode;
  /**
   * 前置内容
   */
  suffixContent?: React.ReactNode;
  /**
   * 导航方向
   * @default horizontal
   */
  direction?: 'horizontal' | 'vertical';
  /**
   * 点击菜单项时触发
   */
  onChange?: (value: string | number) => void;
}

export interface MenuItemProps {
  /**
   * 类名
   */
  className?: string;
  /**
   * 内容
   */
  children?: React.ReactNode;
  /**
   * 自定义样式
   */
  style?: React.CSSProperties;
  /**
   * 单项唯一标识
   * @default 索引值
   */
  value?: string | number;
  /**
   * 点击单项时触发
   */
  onClick?: (value: string | number) => void;
}

export interface MenuGroupProps {
  /**
   * 类名
   */
  className?: string;
  /**
   * 内容
   */
  children?: React.ReactNode;
  /**
   * 自定义样式
   */
  style?: React.CSSProperties;
  /**
   * 组标题
   */
  title?: React.ReactNode
}
