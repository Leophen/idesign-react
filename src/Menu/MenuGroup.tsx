import React, { useContext, useEffect, useState } from 'react';
import classNames from 'classnames';
import './index.scss';
import _ from 'lodash'
import Collapse from '../Collapse';
import Dropdown from '../Dropdown';
import Icon from '../Icon';
import { MenuContext } from './index'
import { MenuGroupProps } from './type';

const MenuGroup: React.FC<MenuGroupProps> = (props) => {
  // 从 Menu Context 注入全局属性
  const context = useContext(MenuContext);
  const newProps = context ? context.inject(props) : props;

  const {
    children,
    className,
    style,
    title,
    // 以下为 context 传入
    index,
    active,
    direction,
    onChange,
    ...restProps
  } = newProps

  // 水平 group
  const [horizonGroupValues, setHorizonGroupValues] = useState<any[]>([])
  const [horizonGroupOptions, setHorizonGroupOptions] = useState<any[]>([])

  const valuesArr: any[] = []
  // 将 Menu.Item 递归转为 Dropdown 可用的数组，同时更新组 value 集
  const getOptions = (childList: any[]) => {
    const result: any[] = []
    childList.map(item => {
      if (!_.isArray(item.props.children)) {
        const itemValue = item.props.value || item.props.children
        valuesArr.push(itemValue)
        result.push({
          content: item.props.children,
          value: itemValue
        })
      } else {
        const itemValue = item.props.value || item.props.title
        valuesArr.push(itemValue)
        result.push({
          content: item.props.title,
          value: itemValue,
          children: getOptions(item.props.children)
        })
      }
    })
    return result
  }

  useEffect(() => {
    if (direction === 'horizontal') {
      setHorizonGroupOptions([...getOptions(children)])
      setHorizonGroupValues([...valuesArr])
    }
  }, [])

  const handleSelectItem = (val: any) => {
    onChange?.(val)
  }

  const [hover, setHover] = useState(false)
  const handleTrigger = (trigger: boolean) => {
    setHover(trigger)
  }

  return (
    <>
      {direction === 'horizontal' ? (
        <Dropdown
          className='i-menu-item__group-dropdown'
          trigger="hover"
          value={active}
          options={horizonGroupOptions}
          onClick={handleSelectItem}
          onTrigger={handleTrigger}
        >
          <div
            className={classNames(
              'i-menu-item__group-wrapper',
              horizonGroupValues.includes(active) && 'i-menu-item__group-active',
              hover && 'i-menu-item__group-hover',
            )}
            {...restProps}
          >
            <span className="i-menu-item__txt">{title}</span>
            <Icon name="ArrowUpBold" size={12} />
          </div>
        </Dropdown>
      ) : (
        <Collapse
          className={classNames(
            'i-menu-item__group',
            className
          )}
          style={style}
          hideBorder
          iconPlacement='right'
          {...restProps}
        >
          <Collapse.Item title={title}>
            {React.Children.map(children, (child) => {
              if (!React.isValidElement(child)) {
                return null;
              }
              const childProps = {
                index: !_.isArray((child as any).props.children) ? (child as any).props.children : (child as any).props.title,
                active,
                onChange,
                ...(child as any).props
              };
              return React.cloneElement(child, childProps);
            })}
          </Collapse.Item>
        </Collapse>
      )}
    </>
  )
}

export default MenuGroup;
