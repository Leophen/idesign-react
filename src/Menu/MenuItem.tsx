import React, { useContext } from 'react';
import classNames from 'classnames';
import './index.scss';
import { MenuContext } from './index';
import { MenuItemProps } from './type';

const MenuItem: React.FC<MenuItemProps> = (props) => {
  // 从 Menu Context 注入全局属性
  const context = useContext(MenuContext);
  const newProps = context ? context.inject(props) : props;

  const {
    children,
    className,
    style,
    value = '',
    onClick,
    // 以下为 context 传入
    active,
    index,
    onChange,
    ...restProps
  } = newProps

  const itemValue = value || index
  const handleClickItem = () => {
    onClick?.(itemValue)
    onChange?.(itemValue)
  }

  return (
    <li
      className={classNames(
        'i-menu-item',
        itemValue === active && 'i-menu-item__active',
        className
      )}
      style={{ ...style }}
      onClick={handleClickItem}
      {...restProps}
    >
      <span className="i-menu-item__txt">{children}</span>
    </li>
  )
}

export default MenuItem;
