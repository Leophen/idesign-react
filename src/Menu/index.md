---
nav:
  title: 组件
  path: /components
group:
  title: 导航组件
  order: 2
order: 1
---

# Menu 导航栏

收纳、排列并展示一系列选项的导航栏。

## 基本用法

使用 `Menu` 包裹菜单的每一项 `Menu.Item`：

```tsx
import React, { useState } from 'react'
import { Menu } from 'idesign-react'

const App = () => {
  const [value, setValue] = useState(1)

  const handleChange = (val) => {
    console.log(val)
    setValue(val)
  }

  return (
    <>
      <h4>无默认值</h4>
      <Menu>
        <Menu.Item>菜单1</Menu.Item>
        <Menu.Item>菜单2</Menu.Item>
        <Menu.Item>菜单3</Menu.Item>
        <Menu.Item>菜单4</Menu.Item>
      </Menu>
      <h4>有默认值</h4>
      <Menu defaultActive={value}>
        <Menu.Item>菜单1</Menu.Item>
        <Menu.Item>菜单2</Menu.Item>
        <Menu.Item>菜单3</Menu.Item>
        <Menu.Item>菜单4</Menu.Item>
      </Menu>
      <h4>有受控值</h4>
      <Menu active={value}>
        <Menu.Item>菜单1</Menu.Item>
        <Menu.Item>菜单2</Menu.Item>
        <Menu.Item>菜单3</Menu.Item>
        <Menu.Item>菜单4</Menu.Item>
      </Menu>
      <h4>一般用法</h4>
      <Menu active={value} onChange={handleChange}>
        <Menu.Item>菜单1</Menu.Item>
        <Menu.Item>菜单2</Menu.Item>
        <Menu.Item>菜单3</Menu.Item>
        <Menu.Item>菜单4</Menu.Item>
      </Menu>
    </>
  )
}

export default App
```

## 带前后缀

通过 `prefixContent` 和 `suffixContent` 分别设置导航项的前后缀：

```tsx
import React from 'react';
import { Menu } from 'idesign-react';

const App = () => {
  const prefix = <img src="/images/logo.svg" />
  const suffix = <div>个人中心</div>

  return (
    <Menu defaultActive='b' prefixContent={prefix} suffixContent={suffix}>
      <Menu.Item value='a'>菜单1</Menu.Item>
      <Menu.Item value='b'>菜单2</Menu.Item>
      <Menu.Item value='c'>菜单3</Menu.Item>
      <Menu.Item value='d'>菜单4</Menu.Item>
    </Menu>
  );
};

export default App;
```

## 嵌套导航

可使用 `Menu.Group` 嵌套多层导航：

```tsx
import React, { useState } from 'react';
import { Menu } from 'idesign-react';

const App = () => {
  const [value, setValue] = useState('2-1')

  const handleChange = (val) => {
    console.log(val)
    setValue(val)
  }

  return (
    <Menu active={value} onChange={handleChange}>
      <Menu.Item value='1'>菜单1</Menu.Item>
      <Menu.Group title='菜单2'>
        <Menu.Item value='2-1'>菜单2-1</Menu.Item>
        <Menu.Group title='菜单2-2'>
          <Menu.Item value='2-2-1'>菜单2-2-1</Menu.Item>
          <Menu.Item value='2-2-2'>菜单2-2-2</Menu.Item>
          <Menu.Item value='2-2-3'>菜单2-2-3</Menu.Item>
        </Menu.Group>
        <Menu.Item value='2-3'>菜单2-3</Menu.Item>
      </Menu.Group>
      <Menu.Item value='3'>菜单3</Menu.Item>
      <Menu.Item value='4'>菜单4</Menu.Item>
    </Menu>
  );
};

export default App;
```

## 垂直方向

通过 `direction` 属性控制导航方向：

```tsx
import React, { useState } from 'react';
import { Menu } from 'idesign-react';

const App = () => {
  const prefix = <img src="/images/logo.svg" />
  const suffix = <div>个人中心</div>

  const [value, setValue] = useState(0)

  const handleChange = (val) => {
    console.log(val)
    setValue(val)
  }

  return (
    <Menu
      width={200}
      direction="vertical"
      active={value}
      prefixContent={prefix}
      suffixContent={suffix}
      onChange={handleChange}
    >
      <Menu.Item>菜单1</Menu.Item>
      <Menu.Group title='菜单2'>
        <Menu.Item>菜单2-1</Menu.Item>
        <Menu.Group title='菜单2-2'>
          <Menu.Item>菜单2-2-1</Menu.Item>
          <Menu.Item>菜单2-2-2</Menu.Item>
          <Menu.Item>菜单2-2-3</Menu.Item>
        </Menu.Group>
        <Menu.Item>菜单2-3</Menu.Item>
      </Menu.Group>
      <Menu.Item>菜单3</Menu.Item>
      <Menu.Item>菜单4</Menu.Item>
    </Menu>
  );
};

export default App;
```

## Menu API

<API hideTitle />

## MenuItem API

| 属性      | 说明           | 类型                       | 默认值   |
| --------- | -------------- | -------------------------- | -------- |
| children  | 按钮内容       | `ReactNode`                | `--`     |
| className | 类名           | `string`                   | `--`     |
| style     | 自定义样式     | `CSSProperties`            | `--`     |
| value     | 单项唯一标识   | `string〡number`           | `索引值` |
| onClick   | 点击单项时触发 | `(string〡number) => void` | `--`     |

## MenuGroup API

| 属性      | 说明       | 类型            | 默认值 |
| --------- | ---------- | --------------- | ------ |
| children  | 按钮内容   | `ReactNode`     | `--`   |
| className | 类名       | `string`        | `--`   |
| style     | 自定义样式 | `CSSProperties` | `--`   |
| title     | 折叠项标题 | `ReactNode`     | `--`   |
