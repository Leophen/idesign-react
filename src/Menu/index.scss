.i-menu {
  position: relative;
  display: flex;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
  height: 64px;
  padding: 0 12px;
  color: var(--i-font-2);
  font-size: 14px;
  background: var(--i-bg);
  &.i-menu__vertical {
    flex-direction: column;
    height: 100%;
    .i-menu-item__wrapper {
      flex-direction: column;
      gap: 8px;
      width: 100%;
    }
    .i-menu-item {
      width: 100%;
      padding: 0 12px;
      &:not(:last-of-type) {
        margin-right: unset;
      }
    }
    .i-menu-item__txt {
      text-align: unset;
    }
    .i-menu__logo,
    .i-menu__operations {
      width: 100%;
      min-height: 40px;
      margin: 0;
    }
    .i-menu__logo {
      margin-top: 16px;
    }
  }
}

.i-menu__logo {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 40px;
  margin-right: 20px;
  margin-left: 8px;
  img {
    max-height: 100%;
  }
}

.i-menu-item__wrapper {
  display: flex;
  flex: 1;
  gap: 12px;
  align-items: center;
  padding: 0;
}

.i-menu-item,
.i-menu-item__group-wrapper {
  display: flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  min-width: 100px;
  height: 40px;
  border-radius: 8px;
  cursor: pointer;
  transition: all 0.2s ease-out;
  &.i-menu-item__group-active,
  &.i-menu-item__active {
    background: var(--i-bg-active);
  }
  &:hover {
    background: var(--i-bg-hover);
  }
}

.i-menu-item__group-dropdown {
  &[data-popper-placement^='bottom']::before {
    top: -17px;
  }
}

.i-menu-item__group-wrapper {
  position: relative;
  .i-menu-item__txt {
    padding: 0 22px;
  }
  .icon-ArrowUpBold {
    position: absolute;
    right: 8px;
    transition: all 0.2s linear;
  }
  &.i-menu-item__group-hover {
    .icon-ArrowUpBold {
      transform: rotateX(180deg);
    }
  }
}

.i-menu-item__txt {
  width: 100%;
  overflow: hidden;
  white-space: nowrap;
  text-align: center;
  text-overflow: ellipsis;
}

.i-menu__operations {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 8px;
  margin-left: 20px;
}

.i-menu-item__group {
  width: 100%;
  .i-collapse-item__header {
    border-radius: 8px;
    &[data-active='1'] {
      .i-collapse-item__header-txt,
      .icon-ArrowCaretRight {
        color: var(--i-primary-txt);
      }
    }
    &:hover {
      background: var(--i-bg-active);
    }
  }
  .i-collapse-item__content {
    &[data-active='1'] {
      margin-bottom: -8px;
    }
    &.i-collapse-item__content-unfold {
      padding: 8px 0 0;
      background: transparent;
    }
  }
  .i-collapse-item__content-inner {
    display: flex;
    flex-direction: column;
    gap: 8px;
    .i-menu-item,
    .i-collapse-item__header {
      padding: 0 12px 0 24px;
    }
    .i-collapse-item__content-inner {
      .i-menu-item,
      .i-collapse-item__header {
        padding: 0 12px 0 48px;
      }
      .i-collapse-item__content-inner {
        .i-menu-item,
        .i-collapse-item__header {
          padding: 0 12px 0 60px;
        }
      }
    }
  }
}
