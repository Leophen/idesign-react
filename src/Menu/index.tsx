import React from 'react';
import classNames from 'classnames';
import './index.scss';
import useDefault from '../hooks/useDefault';
import MenuItem from './MenuItem';
import MenuGroup from './MenuGroup';
import { MenuItemProps, MenuProps } from './type';

export const MenuContext = React.createContext(null as any);

const Menu: React.FC<MenuProps> & {
  Item: React.ElementType,
  Group: React.ElementType
} = (props) => {
  let defaultVal
  React.Children.map(props.children, (child, index) => {
    index === 0 && (defaultVal = (child as any).props.value || 0)
  })

  const {
    children,
    className,
    style,
    width,
    active,
    defaultActive = defaultVal,
    prefixContent,
    suffixContent,
    direction = 'horizontal',
    onChange,
    ...restProps
  } = props;

  const [innerActive, setInnerActive] = useDefault(active, defaultActive, onChange);

  // 注入每一项的 context
  const context = {
    // 将 props 注入每一项子节点的方法
    inject: (singleMenuProps: MenuItemProps) => {
      return {
        direction,
        ...singleMenuProps,
      };
    },
  };

  const handleChange = (val: string | number) => {
    setInnerActive(val)
  }

  return (
    <MenuContext.Provider value={context}>
      <div
        className={classNames(
          'i-menu',
          direction === 'vertical' && 'i-menu__vertical',
          className
        )}
        style={{ ...style, width }}
        {...restProps}
      >
        {prefixContent && (
          <header className="i-menu__logo">
            {prefixContent}
          </header>
        )}
        <ul className="i-menu-item__wrapper">
          {React.Children.map(children, (child, index) => {
            if (!React.isValidElement(child)) {
              return null;
            }
            const childProps = {
              index,
              active: innerActive,
              onChange: handleChange,
              ...child.props
            };
            return React.cloneElement(child, childProps);
          })}
        </ul>
        {suffixContent && (
          <footer className="i-menu__operations">
            {suffixContent}
          </footer>
        )}
      </div>
    </MenuContext.Provider>
  )
};

Menu.Item = MenuItem;
Menu.Group = MenuGroup;

MenuItem.displayName = 'MenuItem';
MenuGroup.displayName = 'MenuGroup';
Menu.displayName = 'Menu';

export default Menu;
