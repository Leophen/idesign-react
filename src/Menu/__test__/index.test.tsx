import { render, fireEvent } from '@testing-library/react';
import Menu from '../index';

describe('Menu 组件测试', () => {
  it('create', () => {
    const { asFragment } = render(
      <Menu>
        <Menu.Item>菜单1</Menu.Item>
        <Menu.Item>菜单2</Menu.Item>
        <Menu.Item>菜单3</Menu.Item>
      </Menu>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('active', () => {
    const { getByTestId } = render(
      <Menu active='1'>
        <Menu.Item>菜单1</Menu.Item>
        <Menu.Item data-testid='test-item'>菜单2</Menu.Item>
        <Menu.Item>菜单3</Menu.Item>
      </Menu>
    );
    expect(getByTestId('test-item')).toHaveClass('i-menu-item__active');
  });

  it('defaultActive', () => {
    const { getByTestId } = render(
      <Menu defaultActive='1'>
        <Menu.Item>菜单1</Menu.Item>
        <Menu.Item data-testid='test-item'>菜单2</Menu.Item>
        <Menu.Item>菜单3</Menu.Item>
      </Menu>
    );
    expect(getByTestId('test-item')).toHaveClass('i-menu-item__active');
  });

  it('onChange', () => {
    const changeFn = jest.fn();
    const { getByTestId } = render(
      <Menu onChange={changeFn}>
        <Menu.Item>菜单1</Menu.Item>
        <Menu.Item data-testid='test-item'>菜单2</Menu.Item>
        <Menu.Item>菜单3</Menu.Item>
      </Menu>
    );
    fireEvent.click(getByTestId('test-item'));
    expect(changeFn).toHaveBeenCalled();
  });

  it('prefixContent suffixContent', () => {
    const prefix = <img src="https://picsum.photos/180/120" />
    const suffix = <div>个人中心</div>

    const { container } = render(
      <Menu prefixContent={prefix} suffixContent={suffix}>
        <Menu.Item>菜单1</Menu.Item>
        <Menu.Item>菜单2</Menu.Item>
        <Menu.Item>菜单3</Menu.Item>
      </Menu>
    );
    expect(container.firstChild.firstChild).toHaveClass('i-menu__logo');
    expect(container.firstChild.lastChild).toHaveClass('i-menu__operations');
  });

  it('Menu Group', () => {
    const { asFragment } = render(
      <Menu>
        <Menu.Item>菜单1</Menu.Item>
        <Menu.Group title='菜单2'>
          <Menu.Item>菜单2-1</Menu.Item>
          <Menu.Group title='菜单2-2'>
            <Menu.Item>菜单2-2-1</Menu.Item>
            <Menu.Item>菜单2-2-2</Menu.Item>
            <Menu.Item value='menu2-2-3'>菜单2-2-3</Menu.Item>
          </Menu.Group>
          <Menu.Item value='menu2-3'>菜单2-3</Menu.Item>
        </Menu.Group>
        <Menu.Item>菜单3</Menu.Item>
      </Menu>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Menu Group width & Direction', () => {
    const { asFragment } = render(
      <Menu width={200} direction="vertical">
        <Menu.Item>菜单1</Menu.Item>
        <Menu.Group title='菜单2'>
          <Menu.Item>菜单2-1</Menu.Item>
          <Menu.Group title='菜单2-2'>
            <Menu.Item>菜单2-2-1</Menu.Item>
            <Menu.Item>菜单2-2-2</Menu.Item>
            <Menu.Item value='menu2-2-3'>菜单2-2-3</Menu.Item>
          </Menu.Group>
          <Menu.Item value='menu2-3'>菜单2-3</Menu.Item>
        </Menu.Group>
        <Menu.Item>菜单3</Menu.Item>
      </Menu>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

