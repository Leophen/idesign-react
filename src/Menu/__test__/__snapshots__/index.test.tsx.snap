// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu 组件测试 Menu Group 1`] = `
<DocumentFragment>
  <div
    class="i-menu"
  >
    <ul
      class="i-menu-item__wrapper"
    >
      <li
        class="i-menu-item i-menu-item__active"
        direction="horizontal"
      >
        <span
          class="i-menu-item__txt"
        >
          菜单1
        </span>
      </li>
      <div
        class="i-popup__reference"
      >
        <div
          class="i-menu-item__group-wrapper"
        >
          <span
            class="i-menu-item__txt"
          >
            菜单2
          </span>
          <i
            class="i-icon icon-ArrowUpBold"
            style="font-size: 12px;"
          />
        </div>
      </div>
      <li
        class="i-menu-item"
        direction="horizontal"
      >
        <span
          class="i-menu-item__txt"
        >
          菜单3
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`Menu 组件测试 Menu Group width & Direction 1`] = `
<DocumentFragment>
  <div
    class="i-menu i-menu__vertical"
    style="width: 200px;"
  >
    <ul
      class="i-menu-item__wrapper"
    >
      <li
        class="i-menu-item i-menu-item__active"
        direction="vertical"
      >
        <span
          class="i-menu-item__txt"
        >
          菜单1
        </span>
      </li>
      <div
        class="i-collapse i-collapse__hide-border i-menu-item__group"
      >
        <div
          class="i-collapse-item i-collapse-item__icon-right"
        >
          <header
            class="i-collapse-item__header i-collapse-item__header-fold"
            data-active="0"
          >
            <i
              class="i-icon icon-ArrowCaretRight"
              style="transform: rotate(0deg);"
            />
            <span
              class="i-collapse-item__header-txt"
            >
              菜单2
            </span>
          </header>
          <section
            class="i-collapse-item__content"
            data-active="0"
            style="height: 0px;"
          >
            <div
              class="i-collapse-item__content-inner"
            >
              <li
                class="i-menu-item"
              >
                <span
                  class="i-menu-item__txt"
                >
                  菜单2-1
                </span>
              </li>
              <div
                class="i-collapse i-collapse__hide-border i-menu-item__group"
              >
                <div
                  class="i-collapse-item i-collapse-item__icon-right"
                >
                  <header
                    class="i-collapse-item__header i-collapse-item__header-fold"
                    data-active="0"
                  >
                    <i
                      class="i-icon icon-ArrowCaretRight"
                      style="transform: rotate(0deg);"
                    />
                    <span
                      class="i-collapse-item__header-txt"
                    >
                      菜单2-2
                    </span>
                  </header>
                  <section
                    class="i-collapse-item__content"
                    data-active="0"
                    style="height: 0px;"
                  >
                    <div
                      class="i-collapse-item__content-inner"
                    >
                      <li
                        class="i-menu-item"
                      >
                        <span
                          class="i-menu-item__txt"
                        >
                          菜单2-2-1
                        </span>
                      </li>
                      <li
                        class="i-menu-item"
                      >
                        <span
                          class="i-menu-item__txt"
                        >
                          菜单2-2-2
                        </span>
                      </li>
                      <li
                        class="i-menu-item"
                      >
                        <span
                          class="i-menu-item__txt"
                        >
                          菜单2-2-3
                        </span>
                      </li>
                    </div>
                  </section>
                </div>
              </div>
              <li
                class="i-menu-item"
              >
                <span
                  class="i-menu-item__txt"
                >
                  菜单2-3
                </span>
              </li>
            </div>
          </section>
        </div>
      </div>
      <li
        class="i-menu-item"
        direction="vertical"
      >
        <span
          class="i-menu-item__txt"
        >
          菜单3
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`Menu 组件测试 create 1`] = `
<DocumentFragment>
  <div
    class="i-menu"
  >
    <ul
      class="i-menu-item__wrapper"
    >
      <li
        class="i-menu-item i-menu-item__active"
        direction="horizontal"
      >
        <span
          class="i-menu-item__txt"
        >
          菜单1
        </span>
      </li>
      <li
        class="i-menu-item"
        direction="horizontal"
      >
        <span
          class="i-menu-item__txt"
        >
          菜单2
        </span>
      </li>
      <li
        class="i-menu-item"
        direction="horizontal"
      >
        <span
          class="i-menu-item__txt"
        >
          菜单3
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
